pkg/updater/resultstore/resultstore.go

```diff
@@ -453,6 +456,85 @@ func processGroup(tg *configpb.TestGroup, group *invocationGroup) *updater.Infla
 	return col
 }
 
+// getTestResults traverses through a test suite and returns a list of all tests
+// that exists inside of it.
+func getTestResults(testSuite *resultstorepb.TestSuite) []*resultstorepb.Test {
+	if testSuite == nil {
+		return nil
+	}
+
+	if len(testSuite.GetTests()) == 0 {
+		return []*resultstorepb.Test{
+			{
+				TestType: &resultstorepb.Test_TestSuite{TestSuite: testSuite},
+			},
+		}
+	}
+
+	var tests []*resultstorepb.Test
+	for _, test := range testSuite.GetTests() {
+		if test.GetTestCase() != nil {
+			tests = append(tests, test)
+		} else {
+			tests = append(tests, getTestResults(test.GetTestSuite())...)
+		}
+	}
+	return tests
+}
+
+func testMethodLimit(tg *configpb.TestGroup) int {
+	var testMethodLimit int
+	const defaultTestMethodLimit = 20
+	if tg == nil {
+		return defaultTestMethodLimit
+	}
+	if tg.EnableTestMethods {
+		testMethodLimit = int(tg.MaxTestMethodsPerTest)
+		if testMethodLimit == 0 {
+			testMethodLimit = defaultTestMethodLimit
+		}
+	}
+	return testMethodLimit
+}
+
+func testResultStatus(testResult *resultstorepb.Test) statuspb.TestStatus {
+	return statuspb.TestStatus(testResult.GetTestCase().Result)
+}
```

Comment:
you have to perform explicit mapping between TestCase results and Testgrid test statuses. e.g.

skipped = pass with skips, suppressed = pass with skips, etc. refer to :


```
func mapStatus(status testsuitepb.TestCase_Result) TestResultStatus {
	switch status {
	case testsuitepb.TestCase_COMPLETED:
		return TestCompleted
	case testsuitepb.TestCase_INTERRUPTED:
		return TestInterrupted
	case testsuitepb.TestCase_CANCELLED:
		return TestCancelled
	case testsuitepb.TestCase_FILTERED:
		return TestFiltered
	case testsuitepb.TestCase_SKIPPED:
		return TestSkipped
	case testsuitepb.TestCase_SUPPRESSED:
		return TestSuppressed
	default:
		return TestResultUnspecified
	}
}
```

and
```
case res == TestSkipped:
							trCell.Result = tspb.TestStatus_PASS_WITH_SKIPS
						case res == TestSuppressed:
							trCell.Result = tspb.TestStatus_PASS_WITH_SKIPS
						case res == TestCancelled:
							trCell.Result = tspb.TestStatus_CANCEL
						case res == TestInterrupted:
							trCell.Result = tspb.TestStatus_CANCEL
						case testResult.Failures() > 0 || testResult.Errors() > 0:
							trCell.Result = tspb.TestStatus_FAIL
						default:
							trCell.Result = tspb.TestStatus_PASS
```

Reply:
also add a couple of test cases to processGroup, that function is affected by your changes now

Reply:
done